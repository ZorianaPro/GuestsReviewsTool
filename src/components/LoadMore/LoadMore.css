.LoadMore .LoadMore-Button {
  cursor: pointer;
  display: flex;
  align-items: center;
  width: 13em;
  justify-content: center;
  margin: 2em auto;
  border: 1px solid var(--colorBlue);
  border-radius: var(--borderRadius);
  transition: all 200ms ease-in;
}

.LoadMore .LoadMore-Button:hover {
  background-color: var(--colorBlue);
  color: var(--colorWhite);
  transition: all 200ms ease-in;
}

.LoadMore .LoadMore-Button p {
  display: inline-block;
}