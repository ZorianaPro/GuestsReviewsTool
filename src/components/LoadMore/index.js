import React, { useState, useEffect, useCallback } from 'react';
import './LoadMore.css';

const LoadMore = ({
  onClick,
  isLoading
}) => {
  const [_isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    setIsLoading(isLoading);
  }, [isLoading]);

  const _onClick = useCallback((event) => {
    if (typeof onClick === 'function') {
      onClick(event);
    }
  }, [onClick]);

  return(
    <div className="LoadMore">
      <div className="LoadMore-Button"
        onClick={ _onClick }
      >
        <p>load more</p>
        {
          _isLoading
          && <span className="LoadMore-Button-Spinner">spinner</span>
        }
      </div>
    </div>
  );
};

export default LoadMore;