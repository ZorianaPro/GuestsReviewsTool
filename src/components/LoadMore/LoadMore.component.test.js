import React from 'react';
import { shallow, mount } from 'enzyme';
import LoadMore from './';
import { act } from 'react-dom/test-utils';

describe('LoadMore', () => {
  it('renders without crashing', () => {
    expect(
      shallow.bind(
        shallow,
        <LoadMore />
      ))
      .not
      .toThrow();
  });

  it('renders as expected', () => {
    expect(
      mount(
        <LoadMore />
      )
    ).toMatchSnapshot();
  });

  describe('on click load button', () => {
    let loadMore;
    let onClick;

    beforeAll(() => {
      onClick = jest.fn();
      loadMore = mount(
        <LoadMore
          onClick={
            onClick
          }
        />
      );
      act(() => {
        loadMore
          .find('.LoadMore-Button')
          .simulate('click');
      });
    });

    it('calls callback', () => {
      expect(onClick)
        .toHaveBeenCalled();
    });
  });

  describe('isLoading', () => {
    describe('is true', () => {
      let loadMore;

      beforeAll(() => {
        loadMore = mount(
          <LoadMore
            isLoading={
              true
            }
          />
        );
      });

      it('shows loading spinner', () => {
        expect(loadMore.exists('.LoadMore-Button-Spinner'))
          .toBe(true);
      });
    });
    describe('is false', () => {
      let loadMore;

      beforeAll(() => {
        loadMore = mount(
          <LoadMore
            isLoading={
              false
            }
          />
        );
      });

      it('shows loading spinner', () => {
        expect(loadMore.exists('.LoadMore-Button-Spinner'))
          .toBe(false);
      });
    });
  });
});
