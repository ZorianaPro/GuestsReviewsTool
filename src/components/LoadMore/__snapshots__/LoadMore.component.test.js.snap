// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMore renders as expected 1`] = `
<LoadMore>
  <div
    className="LoadMore"
  >
    <div
      className="LoadMore-Button"
      onClick={[Function]}
    >
      <p>
        load more
      </p>
    </div>
  </div>
</LoadMore>
`;
